cub/cub/util_type.cuh

```diff
@@ -88,22 +83,11 @@ CUB_NAMESPACE_BEGIN
 #ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
 namespace detail
 {
-
-template <bool Test, class T1, class T2>
-using conditional_t = typename ::cuda::std::conditional<Test, T1, T2>::type;
-
+//! Alias to the given iterator's value_type.
 template <typename Iterator>
-using value_t =
-#  if !defined(_CCCL_COMPILER_NVRTC)
-  typename std::iterator_traits<Iterator>::value_type;
-#  else // defined(_CCCL_COMPILER_NVRTC)
-  typename ::cuda::std::iterator_traits<Iterator>::value_type;
-#  endif // defined(_CCCL_COMPILER_NVRTC)
-
-template <typename It,
-          typename FallbackT,
-          bool = ::cuda::std::
-            is_same<typename ::cuda::std::remove_cv<typename ::cuda::std::remove_pointer<It>::type>::type, void>::value>
+using value_t = typename ::cuda::std::iterator_traits<Iterator>::value_type;
```

Comment:
important: std:: and cuda::std iterator traits are different types. Users are allowed to specialize iterator traits. If they do so on std:: iterator traits, cuda::std:: ones are not affected. This change is potentially breaking. Let's revert this change.

Reply:
Wow, this is horrible! So we should actually never use libcu++'s iterator traits. Or they should query `std::iterator_traits`. In any case, you are right and we should revert this change.

Reply:
> So we should actually never use libcu++'s iterator traits.

We should use cuda::std:: traits in the case of NVRTC simply because there's no other option. NVRTC doesn't ship system headers, so we don't have std:: traits available.